test/CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+cmake_minimum_required(VERSION 3.2.2) 
+
+include(CTest)
+
+
+################################################################################
+# Import Catch Test Framework 
+################################################################################
+
+# See http://dbgit.prakinf.tu-ilmenau.de/code/pfabric/blob/dbe8dbfe21a79c00738cf7762368181edf249767/cmake/Download3rdParty.cmake
+
+include(${CMAKE_SOURCE_DIR}/cmake-helper/DownloadProject/DownloadProject.cmake)
+
+download_project(PROJ               catch
```

Comment:
As much as I like Catch, I don't see how it can be integrated here without rewriting all the tests.

Reply:
IMHO not a big deal. The code is clearly structured. 

Tests not depending on the existence of `sh` is my aim here. I also just wanted to point out how easy it is to incorporate `Catch` with `CMake`, which opens the option for a more flexible CI.